builder/intervals/intervals.go

```diff
@@ -25,17 +24,19 @@ func Load(data []byte) (builder.Intervals, error) {
 	if string(data) == "null" {
 		return i, nil
 	}
-	var t struct {
-		Typ builder.ComponentType `json:"type,omitempty"`
-	}
-	if err := json.Unmarshal(data, &t); err != nil {
+	// "intervals" in the spec is just an string array
+	var intv []string
+	if err := json.Unmarshal(data, &intv); err != nil {
 		return nil, err
 	}
-	switch t.Typ {
-	case "intervals":
-		i = NewIntervals()
-	default:
-		return nil, errors.New("unsupported intervals type")
+	// Now cast the only array item into an actual "interval"
+	interval := Interval(intv[0])
```

Comment:
There's no guarantee that we have an item, it could be `[]`.  We need to be more defensive here - remember, it's user supplied data.

Also, does the spec say it's only one?

Reply:
Unfortunately the Druid documentation for the query spec is... not very specific (from what I've seen).
The `GroupBy` [documentation has](https://druid.apache.org/docs/latest/querying/groupbyquery.html) this to say:
> intervals | A JSON Object representing ISO-8601 Intervals. This defines the time ranges to run the query over.
The code example is a single object array
```
"intervals": [ "2012-01-01T00:00:00.000/2012-01-03T00:00:00.000" ],
```

However some [other documentation](https://druid.apache.org/docs/latest/operations/api-reference.html#metadata-store-information) shows that multiple time ranges are acceptable.

>Returns a list of all segments, overlapping with any of given intervals, for a datasource as stored in the metadata store. Request body is array of string IS0 8601 intervals like [interval1, interval2,...] for example ["2012-01-01T00:00:00.000/2012-01-03T00:00:00.000", "2012-01-05T00:00:00.000/2012-01-07T00:00:00.000"]

Reply:
But this is a good point, the code should be able to handle multiple intervals.